/*
 * Copyright 2012-2019 Johns Hopkins University
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Developed by: Roberts Group
 *               Johns Hopkins University
 *               http://biophysics.jhu.edu/roberts/
 *
 * Author(s): Elijah Roberts, Max Klein
 */

syntax = "proto2";
package lm.io;

import "lm/io/CMEState.proto";
import "lm/io/DiffusionPDEState.proto";
import "lm/io/RDMEState.proto";

message TrajectoryState {

    required uint64 trajectory_id                                 = 1;
    required bool trajectory_started                              = 2;

    optional lm.io.CMEState cme_state                             = 3;
    optional lm.io.RDMEState rdme_state                           = 4;
    optional lm.io.DiffusionPDEState diffusion_pde_state          = 7;

    optional bytes solver_state                                   = 100;
}
